cartographer/pose_graph/node.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_POSE_GRAPH_NODE_H_
+#define CARTOGRAPHER_POSE_GRAPH_NODE_H_
+
+#include "cartographer/pose_graph/proto/node.pb.h"
+
+#include <string>
+#include <vector>
+
+namespace cartographer {
+namespace pose_graph {
+
+// TODO(pifon): Change it to struct { string, timestamp} and introduce the
+// necessary operators.
+using NodeId = std::string;
+
+class Node {
+ public:
+  enum class Parameterization {
+    NONE,
+    YAW_ONLY,
+    CONSTANT_YAW,
+  };
+
+  struct ParameterBlock {
+    // Non-owning ptr to values corresponding to a single parameter block.
+    double* values;
+    // Size of the parameter block.
+    size_t size;
+    Parameterization parameterization;
+  };
+
+  explicit Node(NodeId id, bool constant) : node_id_(id), constant_(constant) {}
+  ~Node() = default;
+
+  proto::Node Serialize() const;
+
+  const NodeId node_id() const { return node_id_; }
+  void set_node_id(const NodeId& id) { node_id_ = id; }
+
+  bool constant() const { return constant_; }
+  void set_constant(bool constant) { constant_ = constant; }
+
+  const std::vector<ParameterBlock>& parameter_blocks() {
```

Comment:
`const`? But since I assume we are going to use this for informing the `PoseGraphOptimizer` we probably need a non-const function returning a non-const vector?

Reply:
fair enough